src/com/amazon/ionschema/IonSchemaSystemLogger.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.ionschema
+
+/**
+ * A callback function for logging by the IonSchemaSystem.
+ *
+ * This allows the IonSchemaSystem implementation to be agnostic to
+ * the actual logging framework used in an application.
+ */
+typealias IonSchemaSystemLogger = (level: LogLevel, message: () -> String) -> Unit
+
+enum class LogLevel {
+    /**
+     * Used for logging general information about the execution of library routines.
+     */
+    Info,
```

Comment:
It's hard for me to imagine a scenario where we'd actually use the Info level. We don't want to give users a bunch of information about the internal execution of the library; we really only want to give them warnings that indicate they should take an action, which fits the Warn level. And as you say in the documentation of the Warn level, there's no level above that because the library would just throw an error in that case.

I recommend removing the concept of LogLevel.